### PR TITLE
 Format patch: fix patch creation when selection order is invalid 

### PR DESCRIPTION
**Pull request by pmiossec**
Fixes :

When selecting 2 revisions in the wrong order (even if it seems the more logical one: from selected first and to selected last), the patchs are not generated and a blank message box is displayed

When selecting more than 2 revisions, patch are generated in the wrong order.

## Proposed changes

- Order of selection of revisions should not have effect on what is generated (patches are now generated in chronological order). Patches are generated successfully in any selection order.
- prevent an empty message box when no file generated (replace by an error message)
- Don't close form when generation failed

### Before

gitextensions_format_patch
And patches are not generated.

### After

* Patches are generated successfully in any selection order.

* New error message (even if I can't display it anymore due to the fix on selection order):

image

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 3.1.0
- Build 9602479c6bb5464dee4add337527ffcf47fb35fc (Dirty)
- Git 2.21.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3394.0
- DPI 192dpi (200% scaling)

### CODE REVIEW

File: GitUI/CommandsDialogs/FormFormatPatch.cs [hunk 1]
```diff
@@ -33,6 +34,8 @@ public partial class FormFormatPatch : GitModuleForm
             new TranslationString("Patch result");
         private readonly TranslationString _noGitMailConfigured =
             new TranslationString("There is no email address configured in the settings dialog.");
+        private readonly TranslationString _failCreatePatch =
+            new TranslationString("Fail to create the patch file(s)");
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormFormatPatch.cs]:
```suggestion
            new TranslationString("Unable to create patch file(s)");
```

Comment (pmiossec) [reply GitUI/CommandsDialogs/FormFormatPatch.cs]: Thanks! Fixed.

**Comment by pmiossec** [comment]
FYI, I have made the PR on v3.1
It seems to be the more logical for me.

**Comment by gerhardol** [comment]
> FYI, I have made the PR on v3.1
It seems to be the more logical for me.

The release/3.1 branch already exists, the standard is to merge to master first, then cherry pick to release if required. I would like to get 3.1 released and merge PRs to master again.

**Comment by RussKie** [comment]
> > FYI, I have made the PR on v3.1
> > It seems to be the more logical for me.
> 
> The release/3.1 branch already exists, the standard is to merge to master first, then cherry pick to release if required. I would like to get 3.1 released and merge PRs to master again.

I prefer the same please.

**Comment by RussKie** [comment]
❗️This fix touches translations, we shouldn't be pulling anything that changes strings in to an upcoming release.
I'm willing to make an exception - there is still a week left before a release, and this error message isn't something a lot of users see.

> > FYI, I have made the PR on v3.1
> > It seems to be the more logical for me.
> 
> The release/3.1 branch already exists, the standard is to merge to master first, then cherry pick to release if required. I would like to get 3.1 released and merge PRs to master again.

I prefer the same.

Please action it quickly.

**Comment by pmiossec** [comment]
️>This fix touches translations, we shouldn't be pulling anything that changes strings in to an upcoming release.
I'm willing to make an exception - there is still a week left before a release, and this error message isn't something a lot of users see.

I know that but it was still made in purpose. With this fix, I don't know if there is a way to see the message. This is just in case because that's better to see a not translated message than an empty message box.

>I prefer the same.

I was not sure of the GE workflow, so I made my branch from the last ancestor commit 😉. 
That way, I was able to easily do my PR from my smartphone!

So, done!

**Pull request merged**